hedera-node/src/main/java/com/hedera/services/ServicesState.java

```diff
@@ -210,6 +219,13 @@ public void init(
             if (SEMANTIC_VERSIONS.deployedSoftwareVersion().isAfter(deserializedVersion)) {
                 migrateFrom(deserializedVersion);
             }
+
+            // Because this flag can be toggled anytime (even post-launch), we need to check for
```

Comment:
I don't think this comment is accurate; if a node reconnected with the `useVirtualMerkle=true` setting, it would migrate independently from the rest of the network and get an ISS.

Reply:
should I guard against cases like this ? My thoughts are that if the flag is flipped post-upgrade (sw version change), we would want to do the migration outside of the standard version upgrade route.